### PR TITLE
fix: correct white routes constant and token extraction gf-13

### PR DESCRIPTION
**Pull request by sandrvvu**
This PR fixes [https://github.com/BinaryStudioAcademy/bsa-2024-gitfit/pull/43](url)
- Adjust white routes constant
- Refactor token extraction to remove 'Bearer ' prefix

### CODE REVIEW

File: apps/backend/src/libs/plugins/authorization/authorization.plugin.ts [hunk 1]
```diff
@@ -28,15 +28,22 @@ const authorization = fp<Options>((fastify, options, done) => {
...
 				status: HTTPCode.UNAUTHORIZED,
 			});
 		}
 
+		const BEARER_PREFIX = "Bearer ";
+		const BEARER_PREFIX_LENGTH = BEARER_PREFIX.length;
+
+		const jwtToken = authHeader.startsWith(BEARER_PREFIX)
+			? authHeader.slice(BEARER_PREFIX_LENGTH)
+			: authHeader;
```
Comment (liza-veis) [inline apps/backend/src/libs/plugins/authorization/authorization.plugin.ts:45]: We can do this easier
```suggestion
		const [, jwtToken] = authHeader.split(BEARER_PREFIX);
```

File: apps/backend/src/libs/plugins/authorization/authorization.plugin.ts [hunk 2]
```diff
@@ -28,15 +28,22 @@ const authorization = fp<Options>((fastify, options, done) => {
 			}
 		}
 
-		const jwtToken = request.headers["authorization"];
+		const authHeader = request.headers["authorization"];
 
-		if (!jwtToken) {
+		if (!authHeader) {
```
Comment (liza-veis) [inline apps/backend/src/libs/plugins/authorization/authorization.plugin.ts:33]: let's move this check here
```suggestion
		if (!authHeader || !authHeader.startsWith(BEARER_PREFIX)) {
```

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: apps/backend/src/libs/plugins/authorization/authorization.plugin.ts [hunk 3]
```diff
@@ -28,15 +28,22 @@ const authorization = fp<Options>((fastify, options, done) => {
...
 			throw new AuthError({
 				message: ExceptionMessage.NO_TOKEN_PROVIDED,
 				status: HTTPCode.UNAUTHORIZED,
 			});
 		}
 
+		const BEARER_PREFIX = "Bearer ";
+		const BEARER_PREFIX_LENGTH = BEARER_PREFIX.length;
+
+		const jwtToken = authHeader.startsWith(BEARER_PREFIX)
```
Comment (GvoFor) [inline apps/backend/src/libs/plugins/authorization/authorization.plugin.ts:43]: I may be wrong, but in our case authorization header always starts with "Bearer ", so there is no need in this check

Comment (liza-veis) [reply apps/backend/src/libs/plugins/authorization/authorization.plugin.ts:43]: We don't know whether this is a request we send from our frontend, so we should check. Also, backend doesn't know anything about how it's implemented on the frontend.

**Review by yevhenii-hladkov — APPROVED** [review]

**Review by liza-veis — APPROVED** [review]

**Pull request merged**